modules/c++/mem/include/mem/SharedPtr.h

```diff
@@ -25,6 +25,12 @@
 
 #include <sys/sys_config.h>
 
+namespace mem
+{
+template<typename T>
```

Comment:
Can you add a comment here explaining why you're doing this  Otherwise we'll forget in three months and delete this.

Reply:
Comment added.